lib/network/kmsAWS/Client.ts

```diff
@@ -0,0 +1,256 @@
+            credentials: {
+                accessKeyId: ak,
+                secretAccessKey: sk,
+            },
+        } : undefined;
+
+        this.client = new KMS({
+            region,
+            endpoint,
+            httpOptions,
+            ...credentials,
+        });
+    }
+
+    get supportsDefaultKeyPerAccount(): boolean {
+        return this._supportsDefaultKeyPerAccount;
+    }
+
+    createBucketKey(bucketName: string, logger: werelogs.Logger, cb: (err: Error | null, keyId?: string) => void): void {
+        logger.debug("AWS KMS: creating encryption key managed at the bucket level", { bucketName });
+        this.createMasterKey(logger, cb);
+    }
+
+    createMasterKey(logger: werelogs.Logger, cb: (err: Error | null, keyId?: string) => void): void {
+        logger.debug("AWS KMS: creating master encryption key");
+        const params = {};
+        this.client.createKey(params, (err: AWSError, data) => {
+            if (err) {
+                const error = _arsenalError(err);
+                logger.error("AWS KMS: failed to create master encryption key", { err });
+                cb(error);
+                return;
+            }
+            logger.debug("AWS KMS: master encryption key created", { KeyMetadata: data?.KeyMetadata });
+            cb(null, data?.KeyMetadata?.KeyId);
+        });
+    }
+
+    destroyBucketKey(bucketKeyId: string, logger: werelogs.Logger, cb: (err: Error | null) => void): void {
+        logger.debug("AWS KMS: deleting encryption key managed at the bucket level", { bucketKeyId });
+        this.deleteMasterKey(bucketKeyId, logger, cb);
+    }
+
+    deleteMasterKey(masterKeyId: string, logger: werelogs.Logger, cb: (err: Error | null) => void): void {
+        logger.debug("AWS KMS: deleting master encryption key", { masterKeyId });
+        const params = {
+            KeyId: masterKeyId,
+            PendingWindowInDays: 7,
+        };
+        this.client.scheduleKeyDeletion(params, (err: AWSError, data) => {
+            if (err) {
+                const error = _arsenalError(err);
+                logger.error("AWS KMS: failed to delete master encryption key", { err });
+                cb(error);
+                return;
+            }
+
+            if (data?.KeyState && data.KeyState !== "PendingDeletion") {
+                const error = _arsenalError("key is not in PendingDeletion state");
+                logger.error("AWS KMS: failed to delete master encryption key", { err, data });
+                cb(error);
+                return;
+            }
+
+            cb(null);
+        });
+    }
+
+    generateDataKey(
+        cryptoScheme: number,
+        masterKeyId: string,
+        logger: werelogs.Logger,
+        cb: (err: Error | null, plainTextDataKey?: Buffer, cipheredDataKey?: Buffer) => void
+    ): void {
+        logger.debug("AWS KMS: generating data key", { cryptoScheme, masterKeyId });
+        assert.strictEqual(cryptoScheme, 1);
```

Comment:
What's this `cryptoScheme`, it's unused ?

Reply:
for future use cases (multiple encryption scheme) and to enforce version control for the encryption scheme, but in the current version, it is simply a validation step.